### DISCUSSION TITLE
Reth error "file client has no tip" while importing blocks

### DISCUSSION THREAD

**Discussion by debjit-bw** [paradigmxyz/reth#12057] [Help] [answered]
I have exported blocks using geth, and am trying to import them in reth. The file contains the RLPs concatenated, but for simplicity i'm just working with a file with 1 RLP.

```sh
./target/debug/reth import genesis.rlp --chain=./chiado_genesis_alloc.json
```

Running the above gives the following error:

```
% ./target/debug/reth import genesis.rlp --chain=./chiado_genesis_alloc.json         
2024-10-24T16:21:13.601843Z  INFO Initialized tracing, debug log directory: /Users/debjit/Library/Caches/reth/logs/chiado
2024-10-24T16:21:13.609322Z  INFO reth 1.0.8-dev (9a732f6b) starting
2024-10-24T16:21:13.620808Z  INFO Opening storage db_path="/Users/debjit/Library/Application Support/reth/chiado/db" sf_path="/Users/debjit/Library/Application Support/reth/chiado/static_files"
2024-10-24T16:21:13.650432Z  INFO Verifying storage consistency.
2024-10-24T16:21:13.798502Z  INFO Consensus engine initialized
2024-10-24T16:21:13.805114Z  INFO Importing chain file chunk
[/Users/debjit/.cargo/git/checkouts/reth-36d3ea1d1152b20c/9a732f6/crates/cli/commands/src/import.rs:96:13] "debug len:" = "debug len:"
[/Users/debjit/.cargo/git/checkouts/reth-36d3ea1d1152b20c/9a732f6/crates/cli/commands/src/import.rs:96:13] file_client.bodies_len() = 0
Error: file client has no tip

Location:
    /Users/debjit/.cargo/git/checkouts/reth-36d3ea1d1152b20c/9a732f6/crates/cli/commands/src/import.rs:98:47
```

I tried generating an RLP file using reth itself, and then manually decoded both the RLPs. Here's the difference:

| RLP from Geth             |  RLP from Reth |
:-------------------------:|:-------------------------:
<img width="640" alt="Screenshot 2024-10-24 at 9 59 38 PM" src="https://github.com/user-attachments/assets/2aab8189-45fc-402e-87c2-353563d2d328"> | <img width="640" alt="Screenshot 2024-10-24 at 10 00 11 PM" src="https://github.com/user-attachments/assets/ceb21466-c463-40cf-bac6-ab83afd7ba3b">

There's the EIP-1559 field in reth that's not in geth, and a couple extra fields vice versa. Can this be causing the issue?

How can I solve this? If it's the EIP-1559 field, I tried setting `londonBlock` in the genesis to a high number, but still the same problem. In any case, can we make the error message better?

Aside:

I am doing [something like this](https://github.com/paradigmxyz/reth/blob/73adc91a0d4c70fa55ad0018541266aa115f9fcd/book/run/sync-op-mainnet.md) to copy over state and blocks from geth and getting them into reth. Since we have 2 extra fields in the header (pre-merge) which aren't in Eth Mainnet, is there a way to import the blocks without validating the hashes (because the default hasher will not include those 2 fields in the hashing)?

**Comment by debjit-bw** [comment]
Additionally, if we want to start the node from a non-genesis block, it is mandatory to import `all` the blocks from genesis (given we `init-state` with the full state)? Is it possible to just do with the last few blocks (sort of like snap sync)?

If it's mandatory to import all blocks, can we
* disable verification? or
* write custom hashing logic?

Because some old blocks have extra fields which will be ignored by reth, resulting in incorrect hashes.

**Comment by mattsse** [comment, marked as answer]
for op-mainnet there's now an option to skip block/receipt import entirely: https://reth.rs/run/sync-op-mainnet.html#minimal-bootstrap-recommended

the --without-ovm option doesn't quite fit for gnosis, but we could rename this.

note that this is only part of the OP CLI: 

https://github.com/paradigmxyz/reth/blob/fa59bd512e877e4677890b5c154b80816c07480e/crates/optimism/cli/src/commands/init_state/mod.rs#L25-L35

**Comment by debjit-bw** [comment]
This sounds like an amazing idea. Can we move this into reth's cli (out of op-specific cli) while making it general purpose?

**Comment by mattsse** [comment]
yep tracking here https://github.com/paradigmxyz/reth/issues/12090

**Comment by debjit-bw** [comment]
thanks a lot ^-^
